File: src/main/java/com/github/squirrelgrip/meter/service/LineProcessor.java
package com.github.squirrelgrip.meter.service;

import com.github.squirrelgrip.meter.domain.ContextKey;

import java.util.Map;

public interface LineProcessor {
    void preProcess(Map<ContextKey, Object> context);
    void process(String[] tokens, Map<ContextKey, Object> context);
}

File: src/main/kotlin/com/github/squirrelgrip/meter/service/LineProcessor.kt
package com.github.squirrelgrip.meter.service

import com.github.squirrelgrip.meter.domain.MeterContext

interface LineProcessor {
    fun process(
        context: MeterContext,
        lineNumber: Int,
        tokens: Array<String>
    )

    fun process(
        context: MeterContext,
        lineNumber: Int,
        line: String
    )
}